tx.go

```diff
@@ -8,6 +8,8 @@ import (
 	"golang.org/x/net/context"
 )
 
+const NEWRELIC_BUFFER = 10000
```

Comment:
That could bottleneck a lot of calls though. What about making the buffer much smaller (5?) and adding some metrics about how much time messages spent waiting in the channel?